crates/viewer/re_view_spatial/src/heuristics.rs

```diff
@@ -28,7 +29,24 @@ pub fn default_visualized_entities_for_visualizer_kind(
 
             if data.preferred_view_kind == Some(visualizer_kind) {
                 let indicator_matching = ctx.indicated_entities_per_visualizer.get(&id)?;
+
+                if indicator_matching
+                    .iter()
+                    .all(|e| suggested_filter.matches(e))
+                {
+                    // We can return early here, because the intersection below would be empty.
+                    return None;
+                }
+
                 let maybe_visualizable = ctx.maybe_visualizable_entities_per_visualizer.get(&id)?;
+
+                if maybe_visualizable
+                    .iter()
+                    .all(|e| suggested_filter.matches(e))
+                {
+                    return None;
+                }
```

Comment:
confused by this. So if either all indicated or all maybe visualizers match are full contained in the `suggested_filter`, then we say no entity is visualized? Sounds rather strange to me, where do I read this wrong? I would have thought that instead we filter down the final flattend result with the `suggested_filter` at the end.

btw. this is the very place that makes me nervous perf wise. We hurl a lot of entities through this place and have one step more there. Well at least it's a bulk operation and the filter not very complex, I have hope that this can be optimized later if it should show up as an issue (plus the more general notion of "don't run all this things every frame 🤨 " which you're right we could use some nudge towards (don't even think it's terribly hard!) 😄

Reply:
Yeah, that makes sense!